Sources/Foundation/NSStringAPI.swift

```diff
@@ -632,6 +632,7 @@ extension StringProtocol {
 
     /// Returns a representation of the string as a C string
     /// using a given encoding.
+    @available(*, deprecated, message: "On platforms without Objective-C autorelease pools, use withCString instead")
```

Comment:
There's an `NSString.cString(using:)` that has the same issue, FWIW

Reply:
I think that we should refer people to use `String.withCString(encodedAs:_)` instead of this message.

Reply:
Sure. There is one place we use this API in keyed archiver that will have to get a follow-up.